dcap/quoteverify/src/collateral.rs

```diff
@@ -58,6 +77,19 @@ impl<T: AsRef<[u8]>> Collateral for Quote3<T> {
         }
         .into_result()?;
 
+        // It shouldn't happen that `tee_qv_get_collateral()` returns success
+        // while keeping the collateral as a null pointer, but we defend against
+        // it since we don't control the C implementation.
+        if sgx_collateral.is_null() {
+            return Err(Error::QuoteLibrary(QlError::NoQuoteCollateralData));
```

Comment:
:shrug: A bit undecided, could make a dedicated error but this error seemed good enough and I don't think this case is likely to be hit in practice as we probably got an error from `tee_qv_get_collateral()`